### PR TITLE
Use a custom security group

### PR DESCRIPTION
**Pull request by arjunmadan**
- creates the group and adds SSH if the group doesn't exist.
- uses an existing group if it does exist.
- falls back to the default group - "service-maker" if no group is specified.

### CODE REVIEW

File: lib/plugins/rest.js [hunk 1]
```diff
@@ -27,6 +27,11 @@ exports.register = function (server, options, next) {
 		throw error;
 	}
 
+	if (process.env.AWS_DEFAULT_SECURITY_GROUP === undefined) {
```
Comment (a-type) [inline lib/plugins/rest.js]: I don't think this needs to be a required parameter. We can provide a hard-coded default, and optionally allow the user to customize the default via the variable.

File: test/unit/plugins/rest_spec.js [hunk 2]
```diff
@@ -36,15 +38,18 @@ describe("The Rest plugin", function () {
...
 		before(function () {
 			server.connection();
+			environment.set("AWS_DEFAULT_SECURITY_GROUP", VALID_SEC_GROUP);
+
 			return server.registerAsync(Rest);
 		});
 
 		after(function () {
-			return server.stopAsync();
+			environment.restore();
```
Comment (a-type) [inline test/unit/plugins/rest_spec.js]: We lost the server stop here, it looks like.

File: test/unit/plugins/rest_spec.js [hunk 3]
```diff
@@ -151,103 +225,171 @@ describe("The Rest plugin", function () {
...
```
Comment (a-type) [inline test/unit/plugins/rest_spec.js]: Lost server stop here as well. Is there a reason for that?

File: test/unit/plugins/rest_spec.js [hunk 4]
```diff
@@ -258,8 +400,8 @@ describe("The Rest plugin", function () {
 		});
 
 		after(function () {
+			environment.restore();
```
Comment (a-type) [inline test/unit/plugins/rest_spec.js:431]: And here. There are more, but I think you get the idea.

File: test/unit/plugins/rest_spec.js (hunk 3 above)
Comment (arjunmadan) [inline test/unit/plugins/rest_spec.js]: I realized we were never actually starting the server.

**Comment by a-type** [comment]
Ok, looks good.

**Pull request merged**